utils/generate-types.js

```diff
@@ -212,14 +212,52 @@ function generateTypeFromTag(param) {
       return normalizeTypeName(param.type.name);
     case 'TypeApplication':
       const baseType = normalizeTypeName(param.type.expression.name);
+      
+      // Handle array cases
+      if (baseType === 'Array') {
+        const innerType = param.type.applications[0];
+        
+        // Handle nested array that represents a tuple
+        if (innerType.type === 'TypeApplication' && 
```

Comment:
Are we able to handle this recursively instead of nesting like this? e.g. something like:
```js
return `(${generateTypeFromTag(innerType)})[]`
```